pkgs/servers/samba/4.x.nix

```
@@ -66,16 +106,20 @@ stdenv.mkDerivation rec {
   configureFlags = [
     "--with-static-modules=NONE"
     "--with-shared-modules=ALL"
-    "--with-system-mitkrb5"
-    "--with-system-mitkdc" krb5Full
     "--enable-fhs"
     "--sysconfdir=/etc"
     "--localstatedir=/var"
     "--disable-rpath"
   ] ++ singleton (if enableDomainController
          then "--with-experimental-mit-ad-dc"
          else "--without-ad-dc")
-    ++ optionals (!enableLDAP) [ "--without-ldap" "--without-ads" ]
+    ++ optionals enableKerberos [
+    "--with-system-mitkrb5"
+    "--with-system-mitkdc ${krb5Full}"
```

Currently this works as `configureFlags` are split by bash but when we switch to proper argument arrays with `structuredAttrs`, it will break.

You shoould use

```suggestion
    "--with-system-mitkdc" "${krb5Full}"
```

or better

```suggestion
    "--with-system-mitkdc=${krb5Full}"
```

if the `configure` script supports it.

---

Fixed